google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -84,15 +89,17 @@
   private final String topicName;
 
   private final BatchingSettings batchingSettings;
+  private final boolean enableMessageOrdering;
 
   private final Lock messagesBatchLock;
-  private MessagesBatch messagesBatch;
+  private final Map<String, MessagesBatch> messagesBatches;
```

Comment:
Perhaps we should consider the benefits of having a `ConcurrentHashMap`, we'd be able to parallelize requests that way.